src/v/datalake/arrow_writing_consumer.cc

```diff
@@ -0,0 +1,270 @@
+#include "datalake/arrow_writing_consumer.h"
+
+#include "datalake/protobuf_to_arrow_converter.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/loop.hh>
+
+#include <arrow/array/builder_binary.h>
+#include <arrow/io/file.h>
+#include <arrow/table.h>
+#include <arrow/type_fwd.h>
+#include <parquet/arrow/writer.h>
+
+#include <exception>
+#include <memory>
+
+datalake::arrow_writing_consumer::arrow_writing_consumer(
+  std::filesystem::path local_file_path, std::string protobuf_schema)
+  : _local_file_path(std::move(local_file_path)) {
+    try {
+        _table_builder = std::make_unique<proto_to_arrow_converter>(
+          protobuf_schema);
+    } catch (const std::exception& e) {
+        // Couldn't build a table builder, fall back to schemaless
+        // TODO: Log this
+    }
```

Comment:
nit: longer term, may want to consider relying on some kind of result (e.g. see base/outcome.h) type and moving this work into some `start()` method, in case the exceptions get too unwieldy